hazelcast/src/main/java/com/hazelcast/map/impl/MapServiceContextImpl.java

```diff
@@ -346,8 +346,8 @@ public void flushMaps() {
     @Override
     public void destroyMap(String mapName) {
         // on LiteMembers we don't have a MapContainer, but we may have a Near Cache and listeners
-        mapNearCacheManager.destroyNearCache(mapName);
         nodeEngine.getEventService().deregisterAllListeners(SERVICE_NAME, mapName);
+        mapNearCacheManager.destroyNearCache(mapName);
```

Comment:
this affected TxnMap, Lite Member and MemberMap tests -> depending on the race we received 1 or 2 invalidations, depending when the listener was deregistered. With this change we should not receive any invalidations, which is not bad, since the Near Cache is completely destroyed anyway.